python/pyiceberg/expressions/visitors.py

```diff
@@ -675,6 +701,24 @@ def visit_less_than_or_equal(self, term: BoundTerm[L], literal: Literal[L]) -> b
 
         return ROWS_MIGHT_MATCH
 
+    def visit_starts_with(self, term: BoundTerm[L], literal: Literal[L]) -> bool:
```

Comment:
Same for the ManifestEvaluator, can you copy the tests from Java: https://github.com/apache/iceberg/blob/master/api/src/test/java/org/apache/iceberg/expressions/TestInclusiveManifestEvaluator.java#L538-L627